presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -273,6 +276,8 @@ public Object optimize(SymbolResolver inputs)
     private class Visitor
             extends AstVisitor<Object, Object>
     {
+        private final Map<NodeRef<Expression>, Type> generatedExpressionTypes = new HashMap<>();
```

Comment:
Curious: Why we need this map now?

Reply:
We have generated Expression during the optimization process which don't have a type (previously, we only generate type mapping before optimization). However, to determine Expression is serializable, we need `Type`.